src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -95,7 +88,7 @@ public JsonNode applyInternal(final JsonNode node) throws JsonPatchException {
             throw new JsonPatchException(BUNDLE.getMessage("jsonPatch.parentNotContainer"));
         }
 
-        if (pathToParent.contains("[?(")) { // json filter result is always a list
+        if (pathDetails.isContainsFiltersOrMultiIndexesNotation()) { // json filter result is always a list
```

Comment:
grammar :)
doesContain instead of isContains

Reply:
Auto generated getter for boolean field. Fixed